docs/use-cases/using-Intel-TDX-and-kata.md

```diff
@@ -0,0 +1,70 @@
+# Kata Containers with TDX
+
+Intel® Trust Domain Extensions (TDX) is introducing new, architectural
+elements to help deploy hardware-isolated, virtual machines (VMs)
+called trust domains (TDs), read
+[Intel® Trust Domain Extensions (Intel® TDX)](https://www.intel.com/content/www/us/en/developer/articles/technical/intel-trust-domain-extensions.html)
+for more information.
+
+## Preconditions
+
+* Intel TDX capable bare metal nodes
+* Host kernel Linux with TDX support(https://github.com/intel/tdx)
+
+```sh
+$ grep TDX /boot/config-`uname -r`
+CONFIG_KVM_INTEL_TDX=y
+```
+
+## Installation
+
+### Kata Containers TDX Guest Kernel
+
+Run the following commands to install guest kernel with TDX support:
+
+```sh
+$ latest=$(curl http://jenkins.katacontainers.io/job/kata-containers-2.0-kernel-tdx-x86_64-nightly/lastSuccessfulBuild/artifact/artifacts/latest)
+$ sudo -E curl -Lk http://jenkins.katacontainers.io/job/kata-containers-2.0-kernel-tdx-x86_64-nightly/lastSuccessfulBuild/artifact/artifacts/vmlinuz-${latest} -o /usr/share/kata-containers/vmlinuz-tdx.container
+```
+
+### Kata Containers TDX QEMU
+
+Run the following command to install QEMU with TDX support:
+
+```sh
+$ curl http://jenkins.katacontainers.io/job/kata-containers-2.0-qemu-tdx-x86_64/lastSuccessfulBuild/artifact/artifacts/kata-static-qemu.tar.gz | sudo tar -C / -zxf -
+```
+
+### Firmware with TDX support
+
+Read [TDVF](https://github.com/tianocore/edk2-staging/tree/TDVF) to build
+and install a firmware with Intel® TDX support
+
+### Kata Containers Configuration
+
+#### Configuration for QEMU
+
+Edit `/usr/share/defaults/kata-containers/configuration.toml` and apply
+the following configuration (look for the sections and change their values accordingly):
+
+```toml
+[hypervisor.qemu]
+kernel = "/usr/share/kata-containers/vmlinuz-tdx.container"
+confidential_guest = true
+kernel_params = "force_tdx_guest tdx_disable_filter"
```

Comment:
documentation doesn't say too much https://github.com/intel/tdx/blob/guest/Documentation/admin-guide/kernel-parameters.txt#L5625

```
tdx_disable_filter [x86]
			Disable TDX guest filter support.
```

:exploding_head:

Reply:
By default TDX guest only allows audited list of drivers/devices and specific list of ACPI tables. This was done to keep the kernel secure from using unsafe drivers. So by default, if you enable some un-audited drivers, it will not work. "tdx_disable_filter" command line options removes this restriction (this is a debug feature to test new drivers in kernel).

Reply:
@sathyaintel - It sounds like `tdx_disable_filter` is a temporary solution until the code lands upstream. Is that correct? Also, what does "audited" mean in this context and do we have thoughts on when the TDX code will be audited?

Reply:
In this context "audit" means driver review. Once the driver is reviewed and confirmed that it has no security implications when working with unstrusted host, it can be added to TDX whitelist (check arch/x86/kernel/tdx-filter.c). Currently, we have audited the virtio PCI drivers and added them in whitelist. Also, the "tdx_disable_filter" command line option is a means for customers to test their drivers (it is a debug option). Once they have a list of drivers they want to whitelist, current suggested approach is to add it via "authorize_allow_devs" command line option or add it directly to the whitelist in arch/x86/kernel/tdx-filter.c

Reply:
Thanks for explaining this @sathyaintel.